lib/runtime/procs/join.js

```diff
@@ -801,28 +801,44 @@ class join extends base {
         return joinpts;
     }
     emit_result(points) {
-        // emit the points, possibly bracketed by marks, and update emitter state.
+        // Emit the points, possibly bracketed by marks, and update emitter state.
+        // Mark interval will be the distance between group start and complete times.
         var leader = this.batched_leader() ;
         var join_time = this.outer ? this.ins[this.outer - 1].live_time() : this.max_live_time();
+        var interval;
+
+        if (leader) {
+            interval = leader.live().complete_time.subtract(leader.live().start_time);
```

Comment:
Thats a good question. I think it might be possible to copy information from mark into the join groups and then extract them after joining.